src/main/java/pl/com/example/ExtractBoolStatementsListener.java

```diff
@@ -2,112 +2,269 @@
+        return ((JavaParser.StatementContext)ctx.parent).start.getType() == IF_TOKEN;
+    }
+
+    /**
+     * Check if met if statement is expanded enough to be extracted to function
+     * @param expression
+     */
+    private boolean isExpandedEnough(String expression){
+
+        Pattern logOpPattern = Pattern.compile(LOG_OP_REGEX);
+        Matcher logOpMatcher = logOpPattern.matcher(expression);
 
-        // Count the occurrences
         int logOpCount = 0;
 
         while (logOpMatcher.find()) {
             logOpCount++;
         }
 
-        if(logOpCount < 2){
-            return;
-        }
+        return logOpCount >= expandedEnoughExpressionIdentifier;
+    }
 
-        RuleContext parent = ctx.parent;
+    /**
+     * Creates boolean function String Template based on functionName and expression
+     * @param functionName
+     * @param expression
+     */
+    private ST createMethodST(String functionName, String expression){
 
-        while(parent.getClass() != JavaParser.MethodBodyContext.class){
+        ST boolMethod = new ST("<line>");
 
-            parent = parent.parent;
+        boolMethod
+                .add(LINE, "\n\n\t")
+                .add(LINE, "private ");
+
+        if(isStatic) {
+            boolMethod.add(LINE, "static ");
         }
 
-        JavaParser.ClassBodyDeclarationContext classBodyDeclarationContext = (JavaParser.ClassBodyDeclarationContext) parent.parent.parent.parent;
+        boolMethod.add(LINE, "boolean " + functionName + "(");
 
-        boolean isStatic = false;
+        List<String> variablesFromExpression = getVariablesListFromExpression(expression);
 
-        for(JavaParser.ModifierContext modifier : classBodyDeclarationContext.modifier()){
-            isStatic = modifier.classOrInterfaceModifier().getText().equals("static");
+        String variableType = null;
+        boolean firstArgumentInserted = false;
 
-            if(isStatic){
-                break;
+        for(String variable : variablesFromExpression){
+
+            variableType = this.symbols.getSymbol(variable);
+
+            if(variableType == null){
+                continue;
             }
-        }
 
-        ST boolMethod = new ST("<line>");
+            if(!firstArgumentInserted){
+                boolMethod.add(LINE, variableType + " " + variable);
+                firstArgumentInserted = true;
+                continue;
+            }
 
-        String funcName = "boolFunc" + funcCounter.toString();
+            boolMethod.add(LINE, ", " + variableType + " " + variable);
+
+        }
 
         boolMethod
-            .add(LINE, "\n\n\t")
-            .add(LINE, "private ");
+                .add(LINE, "){")
+                .add(LINE, "\n\t\t")
+                .add(LINE, "return ")
+                .add(LINE, expression + ";")
+                .add(LINE, "\n\t}");
 
-        if(isStatic) {
-            boolMethod.add(LINE, "static ");
-        }
+        return boolMethod;
+    }
```

Comment:
When variable in expression has no type declaration we should check if it's String/int/float and so on. Instead of a variable there could be a literal.